llvm_util/utils.cpp

```diff
@@ -521,4 +525,67 @@ llvm::Function *findFunction(llvm::Module &M, const string &FName) {
   return F && !F->isDeclaration() ? F : nullptr;
 }
 
+TailCallInfo parse_tail_call_info(const llvm::CallInst &i) {
```

Comment:
we don't encode semantics like this. The semantics must be in the ir folder.
We support other IRs -> Alive2 IR, hence we do as little as possible in the translation.

Reply:
I left `parse_tail_call_info` in utils.cpp so that it may be used from known_functions too (originally was in llvm2alive). I think we still need it there as we're parsing data from LLVM (like `parse_fn_attrs`). Do you mean that the semantics in `are_tailcall_preconditions_met` is to be moved in ir inside `check_tailcall`?

Reply:
Also made it more coherent (`parse_tail_call_info` -> `parse_fn_tailcall`).